### PR TITLE
feat: Kubernetes integration

### PR DESCRIPTION
**Pull request by snowpeakz**

### CODE REVIEW

File: k8s/src/bin/operator.rs [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+
+            job_resources.apply().await?;
+
+            Ok(ReconcilerAction {
+                requeue_after: Some(Duration::from_secs(10)),
+            })
+        }
+        Action::Delete => {
+            let job_name = job.name();
+            eprintln!("Deletding PersiaJob: {}", job_name);
```
Comment (williamstar) [inline k8s/src/bin/operator.rs:89]: typo

File: k8s/src/bin/server.rs [hunk 2]
```diff
@@ -0,0 +1,229 @@
...
+        kubernetes_client,
+    );
+    let resp = match resources.apply().await {
+        Ok(_) => ExecutionResults {
+            success: true,
+            err_msg: None,
+        },
+        Err(e) => ExecutionResults {
+            success: false,
+            err_msg: Some(format!("{:?}", e)),
```
Comment (williamstar) [inline k8s/src/bin/server.rs:66]: `to_string` is simple than `format`

File: k8s/src/crd.rs [hunk 3]
```diff
@@ -0,0 +1,516 @@
...
+#[derive(CustomResource, Serialize, Deserialize, Debug, PartialEq, Clone, JsonSchema)]
+#[kube(
+    group = "persia.com",
+    version = "v1",
+    kind = "PersiaJob",
+    plural = "persiajobs",
+    derive = "PartialEq",
+    namespaced
+)]
+#[allow(non_snake_case)]
```
Comment (williamstar) [inline k8s/src/crd.rs:51]: why non_snake_case? json_schema can't be parse?

File: k8s/src/crd.rs [hunk 4]
```diff
@@ -0,0 +1,516 @@
...
+
+                    container.image = trainer.image.clone();
+                    if container.image.is_none() {
+                        container.image = Some(String::from(DEFAULT_CUDA_IMAGE));
+                    }
+
+                    let env = container
+                        .env
+                        .as_mut()
+                        .expect("no env in a persia podspec template");
```
Comment (williamstar) [inline k8s/src/crd.rs:338]: for `trainer` or `nn_worker` env spec

File: k8s/src/crd.rs (hunk 3 above)
Comment (snowpeakz) [reply k8s/src/crd.rs:51]: To be consistent with k8s, e.g. https://kubernetes.io/docs/concepts/storage/volumes/#aws-ebs-configuration-example

File: k8s/src/bin/operator.rs (hunk 1 above)
Comment (snowpeakz) [reply k8s/src/bin/operator.rs]: Fixed in https://github.com/PersiaML/PERSIA/pull/142/commits/4b7070554666fd7c94ef897a0160e2031c1ab7f0

File: k8s/src/bin/server.rs (hunk 2 above)
Comment (snowpeakz) [reply k8s/src/bin/server.rs]: Fixed in https://github.com/PersiaML/PERSIA/pull/142/commits/4b7070554666fd7c94ef897a0160e2031c1ab7f0

File: k8s/src/crd.rs (hunk 4 above)
Comment (snowpeakz) [reply k8s/src/crd.rs]: Fixed in https://github.com/PersiaML/PERSIA/pull/142/commits/4b7070554666fd7c94ef897a0160e2031c1ab7f0

**Review by NOBLES5E — APPROVED** [review]

**Review by williamstar — APPROVED** [review]
LGTM

**Pull request merged**